newbank/server/Commands/NewAccountCommand.java

```diff
@@ -15,21 +16,30 @@ public String getCommandName() {
 
   @Override
   public String getDescription() {
-    return "<account type> <optional: account name> <optional: currency> " + System.lineSeparator()
-        + "-> Creates a new account of specified type e.g. NEWACCOUNT \"Savings Account\" \"my savings\" EUR. " + System.lineSeparator()
+    return "<account type> <optional: account name> <optional: currency> "
+        + System.lineSeparator()
+        + "-> Creates a new account of specified type e.g. NEWACCOUNT \"Savings Account\" \"my savings\" EUR. "
+        + System.lineSeparator()
         + "   Standard currency is GBP, please specify an account name and currency to create an account with a different currency.";
   }
 
   @Override
   public NewBankCommandResponse run(NewBankCommandParameter param) {
-
     var args = NewAccountCommandArgument.parse(param);
 
-    if (args == null) return NewBankCommandResponse.invalidRequest("FAIL");
+    if (args == null)
+      return NewBankCommandResponse.invalidRequest(
+          "FAIL: Account type must be specified. Accepted account types: "
+              + AccountTypeInfo.listAllAccountTypesCommaDelimited()
+              + ".");
+
+    // Previously this was tested in NewBankArgument
+    if (param.getCustomer().hasAccount(args.getAccountType(), args.getAccountName()))
```

Comment:
Hi Thomas, I think this should use the other hasAccount overload that doesn't require the account type to also match.